test/integration/model/cte.test.js

```diff
@@ -0,0 +1,766 @@
+'use strict';
+
+var chai = require('chai')
+  , expect = chai.expect
+  , things = require('chai-things')
+  , Support = require(__dirname + '/../support')
+  , DataTypes = require(__dirname + '/../../../lib/data-types')
+  , current = Support.sequelize;
+
+  chai.use(things);
+
+
+
+describe(Support.getTestDialectTeaser('CTEs'), function () {
+    if (current.dialect.supports.cteQueries) {
+        describe('with findAll', function () {
+                it('can follow a basic \'has one\' association', function () {
+                    var User = this.sequelize.define('UserXYZ', { username: DataTypes.STRING });
+
+                    User.hasOne(User, { foreignKey: 'manager_id', as: 'report' });
+
+                    return this.sequelize.sync({ force: true }).then(function () {
+                        return User.create({username: 'user1' }).then(function (user1) {
+                            return User.create({username: 'user2' }).then(function (user2) {
+                                return User.create({ username : 'user3'}).then(function(user3) {
+                                    return user3.setReport(user1).then(function () {
+                                        return User.findAll({
+                                            cte: [{
+                                                name: 'a',
+                                                model: User,
+                                                initial: { where: { username: 'user3' } },
+                                                recursive: { next: 'report' }
+                                            }],
+                                            cteSelect: 'a'
```

Comment:
Should we support sugar that allows a single `cte` object (no array) which will become the cteSelect automatically?